### PR TITLE
Increase coverage of api/jobs/handlers.py

### LINKED ISSUE #683: Increase test coverage of api/jobs/handlers.py
Objective is 100% coverage

### PR DESCRIPTION
**Pull request by ambrussimon-invenshure**
Closes #683

8 LOC left in api/jobs/handlers.py (97%)
- 5L Permission handling (that'll be faster to handle globally in #702): 244, 351, 354, 416-417
- 2L Checks that seemed unreachable (maybe obsolete or useful for compatibility?): 166, 222
- 1L That I would rather solve later with additional fixture(s) (need a session w/o acquisitions): 340

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

File: test/integration_tests/abao/abao_test_hooks.js [hunk 1]
```diff
@@ -83,6 +83,7 @@ hooks.skip("POST /projects/{ProjectId}/recalc -> 200")
 
 // Porting to python as per #600
 hooks.skip("POST /jobs/add -> 200")
+hooks.skip("PUT /jobs/{JobId} -> 200")
```
Comment (ambrussimon-invenshure) [inline test/integration_tests/abao/abao_test_hooks.js]: - Added python test for jobs/{JobId}/cancel
- There's no delete for jobs
- Abao uses the first job from GET /jobs
- Cannot modify canceled job

It was quicker to skip it (it's already covered in python) than to raw delete it and/or getting Abao to use a different job.

**Comment by kofalt** [comment]
LGTM! Nice stuff.

**Comment by kofalt** [comment]
@ambrussimon-invenshure This looks green, is there more to do here yet? If you want, rebase or merge master into this branch, see if CI is still green. I'm fine landing this as-is if it passes.

**Comment by ambrussimon-invenshure** [comment]
@kofalt thanks, will merge as soon as CI passes after the rebase! (Was hesitant to merge as didn't hear back from @ryansanford, last time he wanted to make sure multiple reviews happen before that.)

**Comment by ryansanford** [comment]
@ambrussimon-invenshure My apologies. I had typed a LGTM, but never hit the comment button.

**Pull request merged**